synapse/federation/sender/__init__.py

```diff
@@ -536,8 +536,7 @@ async def handle_room_events(events: List[EventBase]) -> None:
 
                 if event_entries:
                     now = self.clock.time_msec()
-                    last_id = next(reversed(event_ids))
-                    ts = event_to_received_ts[last_id]
+                    ts = max(t for t in event_to_received_ts.values() if t)
                     assert ts is not None
```

Comment:
Suspect the assertion is now redundant: max of an empty sequence is a value error, and otherwise it's a maximum of at least one int.